src/frontend/apps/impress/src/features/docs/doc-header/components/DocToolBox.tsx

```diff
@@ -125,6 +147,74 @@ export const DocToolBox = ({ doc, versionId }: DocToolBoxProps) => {
                 {t('Delete document')}
               </Button>
             )}
+            <Button
+              onClick={() => {
+                setIsDropOpen(false);
+                getDocContentFormatted('markdown').then(
+                  (docContentFormatted) => {
+                    navigator.clipboard.writeText(docContentFormatted).then(
+                      () =>
+                        toast(t('Copied to clipboard'), VariantType.SUCCESS, {
+                          duration: 3000,
+                        }),
+                      () =>
+                        toast(
+                          t('Failed to copy to clipboard'),
+                          VariantType.ERROR,
+                          { duration: 3000 },
+                        ),
+                    );
+                  },
+                  () =>
+                    toast(
+                      t('Editor') + ' ' + t('unavailable'),
+                      VariantType.ERROR,
+                      {
+                        duration: 3000,
+                      },
+                    ),
+                );
+              }}
```

Comment:
All this part are similar for the 2 buttons, this code could go in `getDocContentFormatted`, if `getDocContentFormatted` manages the copy, maybe it can be renamed to something more accurate, `copyToClipboard` ?

Reply:
I abstracted the function to get the formatted editor content away to `src/features/docs/doc-header/utils.ts`
And cleaned up the remaining code to avoid duplicated code